python_legacy/iceberg/api/types/conversions.py

```diff
@@ -90,11 +99,7 @@ def to_byte_buffer(type_id, value):
 
     @staticmethod
     def from_byte_buffer(type_var, buffer_var):
-        return Conversions.internal_from_byte_buffer(type_var.type_id, buffer_var)
-
-    @staticmethod
-    def internal_from_byte_buffer(type_id, buffer_var):
         try:
-            return Conversions.from_byte_buff_mapping.get(type_id)(type_id, buffer_var)
+            return Conversions.from_byte_buff_mapping.get(type_var.type_id)(type_var, buffer_var)
         except KeyError:
-            raise TypeError("Cannot deserialize Type: %s" % type_id)
+            raise TypeError("Cannot deserialize Type: %s" % type_var)
```

Comment:
Should this be `type_var.type_id` instead of `type_var`? And maybe an f-string: `f"Cannot deserialize Type: {type_var.type_id}"`

Reply:
I'm not sure what we want to express here.  Here's what I think:
```Python
raise TypeError("Cannot deserialize Type: %s" % type_var)
```
If we want to express that we could not deserialize a type, it makes sense that the error specifies which type it is by echoing what was passed to the function.  It could not deserialize **this** type and not a TypeID.  The dev never asked for a TypeID to be deserialized...

```Python
raise TypeError("Cannot deserialize Type: %s" % type_var.type_id)
```
If we want to use `type_id` in the message, we should say that we couldn't find a converter or a deserializer for this TypeID.  

I think what is causing confusion is that `to_byte_buffer()` and `from_byte_buffer()` don't align...  But maybe it's just me 😄   That's my 2cents, let me know what you prefer.

Reply:
Yes, it's a KeyError that we are catching, but we are raising a TypeError!  Maybe we should do a combination of both.  Let me try something and you can comment on it.

Reply:
Ok, I have changed my mind after make the modifications.  The reason is that for some types like `struct`, the `__str__` method may spit out a very long string.  It goes and [loops over every field](https://github.com/apache/iceberg/blob/master/python_legacy/iceberg/api/types/types.py#L526)...  For this reason, I would refrain from using `type_var` in the exception message to keep logs clean.  Let me know if you diasgree.